R/zchunk_LA1321.Cement_china.R

```diff
@@ -0,0 +1,117 @@
+#' module_gcam.china_LA1321.Cement
+#'
+#' Allocate across the provinces national cement production, input-output cofficients, and energy inputs to cement production
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L1321.out_Mt_province_cement_Yh}, \code{L1321.IO_GJkg_province_cement_F_Yh}, \code{L1321.in_EJ_province_cement_F_Y}. The corresponding file in the
+#' original data system was \code{LA1321.Cement.R} (gcam-china level1).
+#' @details The tables for cement production, i.e., out, and energy inputs, i.e., in, were calculated by applying state shares to national data.
+#' @details The state shares were determined by the states' relative values of cement shipments.
+#' @details The input-out coefficients were downscaled to the states in proportation to the national data.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author Yang Aug 2018
+module_gcam.china_LA1321.Cement <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c("L101.inNBS_Mtce_province_S_F",
+             "L1321.out_Mt_R_cement_Yh",
+             "L1321.IO_GJkg_R_cement_F_Yh",
+             "L1321.in_EJ_R_cement_F_Y"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L1321.out_Mt_province_cement_Yh",
+             "L1321.IO_GJkg_province_cement_F_Yh",
+             "L1321.in_EJ_province_cement_F_Y"))
+  } else if(command == driver.MAKE) {
+
+    all_data <- list(...)[[1]]
+
+    sector <- fuel <- province <- multiplier <- GCAM_region_ID <- year <- value <- NULL   # silence package check notes
+
+    # Load required inputs
+    L1321.out_Mt_R_cement_Yh <- get_data(all_data, "L1321.out_Mt_R_cement_Yh")
+    L1321.IO_GJkg_R_cement_F_Yh <- get_data(all_data, "L1321.IO_GJkg_R_cement_F_Yh")
+    L1321.in_EJ_R_cement_F_Y <- get_data(all_data, "L1321.in_EJ_R_cement_F_Y")
+
+    # Proxy for downscaling cement to provinces
+    # TODO: Find a better proxy
```

Comment:
Please move this to a repo issue and delete this comment (or if you want to keep, reference the issue number).